mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/gitlab/core/GitlabRepositoryConnector.java

```diff
@@ -0,0 +1,544 @@
+	    .build(new CacheLoader<String, byte[]>() {
+
+		@Override
+		public byte[] load(String key) throws Exception {
+		    byte[] avatarBytes = null;
+		    HttpURLConnection connection;
+
+		    connection = (HttpURLConnection) new URL(key).openConnection();
+		    connection.setConnectTimeout(30000);
+		    connection.setUseCaches(false);
+		    connection.connect();
+
+		    if (connection.getResponseCode() == 200) {
+			ByteArrayOutputStream output = new ByteArrayOutputStream();
+			InputStream input = connection.getInputStream();
+			try {
+			    byte[] buffer = new byte[8192];
+			    int read = -1;
+			    while ((read = input.read(buffer)) != -1) {
+				output.write(buffer, 0, read);
+			    }
+			} finally {
+			    try {
+				input.close();
+			    } catch (IOException ignore) {
+			    }
+			}
+			avatarBytes = output.toByteArray();
+		    }
+		    return avatarBytes;
+		}
+	    });
+
+    public GitlabRestClient createClient(TaskRepository repository) {
+	RepositoryLocation location = new RepositoryLocation(convertProperties(repository));
+	AuthenticationCredentials credentials1 = repository
+		.getCredentials(org.eclipse.mylyn.commons.net.AuthenticationType.REPOSITORY);
+	UserCredentials credentials = new UserCredentials(credentials1.getUserName(), credentials1.getPassword(), null,
+		true);
+	location.setCredentials(AuthenticationType.REPOSITORY, credentials);
+	GitlabRestClient client = new GitlabRestClient(location, this, repository);
+
+	return client;
+    }
+
+    private Map<String, String> convertProperties(TaskRepository repository) {
+	return repository.getProperties().entrySet().stream()
+		.collect(Collectors.toMap(e -> convertProperty(e.getKey()), Map.Entry::getValue));
+    }
+
+    @SuppressWarnings("restriction")
+    private String convertProperty(String key) {
+	if (TaskRepository.PROXY_USEDEFAULT.equals(key)) {
+	    return RepositoryLocation.PROPERTY_PROXY_USEDEFAULT;
+	} else if (TaskRepository.PROXY_HOSTNAME.equals(key)) {
+	    return RepositoryLocation.PROPERTY_PROXY_HOST;
+	} else if (TaskRepository.PROXY_PORT.equals(key)) {
+	    return RepositoryLocation.PROPERTY_PROXY_PORT;
+	}
+	return key;
+    }
+
+    @Override
+    public TaskMapper getTaskMapping(final TaskData taskData) {
+
+	return new TaskMapper(taskData) {
+	    @Override
+	    public String getTaskKey() {
+		TaskAttribute attribute = getTaskData().getRoot()
+			.getAttribute(GitlabTaskSchema.getDefault().TASK_KEY.getKey());
+		if (attribute != null) {
+		    return attribute.getValue();
+		}
+		return super.getTaskKey();
+	    }
+
+	    @Override
+	    public String getTaskKind() {
+		return taskData.getConnectorKind();
+	    }
+	};
+    }
+
+    public byte[] getAvatarData(String url) {
+	try {
+	    return avatarCache.get(url);
+	} catch (ExecutionException e) {
+//	    throw new CoreException(
+//		    new Status(IStatus.ERROR, GitlabCoreActivator.PLUGIN_ID, "TaskRepositoryManager is null"));
+	}
+	return null;
```

Comment:
`null` for array should be avoided

Reply:
I reworked that part for the next commit